src/lib/FlightTasks/tasks/Utility/StraightLine.cpp

```diff
@@ -0,0 +1,214 @@
+ *
+ ****************************************************************************/
+
+/**
+ * @file StraightLine.cpp
+ */
+
+#include "StraightLine.hpp"
+#include <mathlib/mathlib.h>
+#include <float.h>
+
+#define ACC_RAD_ZERO_VEL 2.0f
+#define VEL_ZERO_THRESHOLD 0.001f
+#define DECELERATION_MAX 8.0f
+
+using namespace matrix;
+
+StraightLine::StraightLine(ModuleParams *parent, const float &deltatime, const matrix::Vector3f &pos) :
+	ModuleParams(parent),
+	_deltatime(deltatime), _pos(pos)
+{
+
+}
+
+void StraightLine::generateSetpoints(matrix::Vector3f &position_setpoint, matrix::Vector3f &velocity_setpoint)
+{
+	// Check if target position has been reached
+	if (_is_target_reached || (_desired_speed_at_target < VEL_ZERO_THRESHOLD &&
+				   (_pos - _target).length() < ACC_RAD_ZERO_VEL)) {
+		// Vehicle has reached target. Lock position
+		position_setpoint = _target;
+		velocity_setpoint = Vector3f(0.0f, 0.0f, 0.0f);
+		_is_target_reached = true;
+
+		return;
+	}
+
+	// unit vector in the direction of the straight line
+	Vector3f u_orig_to_target = (_target - _origin).unit_or_zero();
+	// vector from origin to current position
+	Vector3f orig_to_pos = _pos - _origin;
+	// current position projected perpendicularly onto desired line
+	Vector3f closest_pt_on_line = _origin + u_orig_to_target * (orig_to_pos * u_orig_to_target);
+	// previous velocity in the direction of the line
+	float speed_sp_prev = math::max(velocity_setpoint * u_orig_to_target, 0.0f);
+
+	// Calculate braking distance depending on speed, speed at target and deceleration (add 10% safety margin)
+	float braking_distance = 1.1f * (powf(_desired_speed - _desired_speed_at_target, 2) / (2.0f * _desired_deceleration));
+
+	float dist_to_target = (_target - _pos).length(); // distance to target
+
+	// Either accelerate or decelerate
+	float speed_sp    = dist_to_target > braking_distance ? _desired_speed        :  _desired_speed_at_target;
+	float max_acc_dec = dist_to_target > braking_distance ? _desired_acceleration : -_desired_deceleration;
+
+	float acc_track = (speed_sp - speed_sp_prev) / _deltatime;
+
+	if (fabs(acc_track) > fabs(max_acc_dec)) {
+		// accelerate/decelerate with desired acceleration/deceleration towards target
+		speed_sp = speed_sp_prev + max_acc_dec * _deltatime;
+	}
+
+	// constrain the velocity
+	speed_sp = math::constrain(speed_sp, 0.0f, _desired_speed);
+
+	// set the position and velocity setpoints
+	position_setpoint = closest_pt_on_line;
+	velocity_setpoint = u_orig_to_target * speed_sp;
+
+}
+
+float StraightLine::getMaxAcc()
+{
+	// unit vector in the direction of the straight line
+	Vector3f u_orig_to_target = (_target - _origin).unit_or_zero();
+
+	// calculate the maximal horizontal acceleration
+	float divider = (sqrt(powf(u_orig_to_target(0), 2) + powf(u_orig_to_target(1), 2)));
+	float max_acc_hor = MPC_ACC_HOR_MAX.get();
+
+	if (divider > FLT_EPSILON) {
+		max_acc_hor /= divider;
+
+	} else {
+		max_acc_hor *= 1000.0f;
```

Comment:
why times 1000? Why not just leave it as MPC_ACC_HOR?

Reply:
`1000` is a arbitrary... That case just means that this direction is zero and therefore I want to have the other acc_max which then needs to be small because of the `min(max_acc_hor, max_acc_vert)` at the end